### PR TITLE
Update l2 api

### PR DESCRIPTION
**Pull request by lmoe**
# Description of change

* Updates the EVM toolkit
* Adds dockerized apiclient generator and updates Makefile `make apiclient-docker` 
* Restricts deletion of the last user
* Sanitizes users permissions
* Shows own node in node list as always trusted

### CODE REVIEW

File: packages/users/user_manager.go [hunk 1]
```diff
@@ -25,6 +30,22 @@ func NewUserManager(storeCallback func([]*User) error) *UserManager {
...
+	sanitizedPermissions := map[string]struct{}{}
+
+	for _, permission := range maps.Keys(permissions) {
+		if isPermissionAllowed(permission) {
+			sanitizedPermissions[permission] = struct{}{}
+		}
+	}
+
+	return sanitizedPermissions
+}
```
Comment (jorgemmsilva) [inline packages/users/user_manager.go:47]: I'm a bit confused here, why is this a `map[string]struct{}`, and why does the sanitized version is just  map of strings to empty stucts?

File: packages/webapi/interfaces/interfaces.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ var (
 	ErrNoBalanceOnAccount     = errors.New("no balance on account")
 	ErrInvalidNonce           = errors.New("invalid nonce")
 	ErrChainNotFound          = errors.New("chain not found")
+	ErrCantDeleteLastUser     = errors.New("you can't delete the last user")
```
Comment (jorgemmsilva) [inline packages/webapi/interfaces/interfaces.go:33]: this shouldn't be needed, since a user can't delete himself, right?

**Review by jorgemmsilva — APPROVED** [review]

File: packages/webapi/interfaces/interfaces.go (hunk 2 above)
Comment (lmoe) [reply packages/webapi/interfaces/interfaces.go:33]: True, unless the authentication scheme is set to `none` and there is no "user" to be checked against.

File: packages/users/user_manager.go (hunk 1 above)
Comment (lmoe) [reply packages/users/user_manager.go:47]: That's how the user manager was built. The emptry struct could also have been a boolean but that was not my decision. 

I just sanitize the map properly so you can't add some bogus permissions to the user. :)

**Pull request merged**